### DISCUSSION TITLE
Electricity based dungeon

### DISCUSSION THREAD

**Discussion by IQGobo** [fornclake/TetraForce#260] [Ideas]
A more advanced dungeon, as rooms could feel very differently when powered on or off. Lights could turn on with the power or change color and traps could be active or inactive to reflect different states of the dungeon progression. Maybe working with electricity could require basic or insulated gloves, otherwise the player gets hurt when in contact?

This dungeon features several objects that should be unique to the area:

- [ ] Generators, that have to be switched on or off, depending on the puzzle requirement. A running generator should glow or spark some particles and be audible. A variant of the generator would be a collectible battery item and stationary battery socket. Those would act as keys and locks, but with the twist, that a battery could be removed from a socket and used in another one, instead of being permanently used up on insertion. Maybe batteries and sockets would allow isolated puzzle rooms, while a generator would provide electricity for the whole dungeon and power on wires in all of its rooms? That would allow us to have small rooms in different parts of the map that you would need to collect batteries for to power up and claim treasures. Maybe you can find dead batteries in the world and have them get charged once you powered the generator in this dungeon?
- [ ] some kind of "wire layer" on the floor. It should visualize the connection just like the generators, with some humming or buzzing sounds and a distinct glow or sparkle when active. Wires should have a cut or broken tile variant, where a player has to bridge the gap to make the current flow.
- [ ] moving conductive blocks, like metal cubes. Those special push blocks would allow for for simple "move it here once and you are done" type puzzles, as well as "move the block away to cut the flow again" or "move the block between several wire gaps to power different gizmos"
- [ ] switches of various kinds: a simple on/off switch, or a switch that would direct the current to either the left or the right wire or a "T" shaped rotary switch, that could connect two or three wires depending on its position.
- [ ] electricity based traps. That could be a simple static laser blocking a way (with a wire running up to it to visualize how it might be turned off) or a stationary rotating laser sweep enemy that can be toggled on or off by wire (imagine a dull orb resting dead on the ground when powered off, but floating a bit above the floor and rotating when powered on). There could also be floor tiles that get charged on a timer and grill a player on it while active, like a spear trap would hurt a player.
- [ ] mechanical or robotic enemies that only move and interact with the player if a room or a wire to their initial position is powered on. They should stop attacking and chasing a player and return to their home position if the power is cut (or simply freeze in place at least).

**Comment by IQGobo** [comment]
Some more thoughts on the charged batteries and sockets to plug them into:

- only charged batteries can be plugged into a socket to power a device
- removing a battery from a socket depletes a certain amount of charge. This would allow us to implement a basic difficulty selection option: hard mode would drain all the power from a battery, while other difficulties would allow two or more uses until the battery needs to get charged again.
- there should either be a single charging station relatively central to the over-world or just a few chargers at strategic locations
- batteries would work like a bomb: it is an active use item with a max amount of carry capacity. Players can "use" batteries on a socket to power a device, charge the battery, or anywhere on the ground to just put them down. Placed batteries need to be a persistent item, it should remain in place even if all players leave the room.
- players can pick up batteries if they are not at their max carry capacity already.
- batteries could be used to power up teleports or doors or "hard light bridges" (walk-able force fields) on the world map. Each teleport would require a battery to stay active. This would allow players to temporarily power certain machinery for a one time access, or leave them running as long as no one removes the battery. Bridges would open optional paths to make world traversal less dependent on certain other items (e.g. you could usually get to a location using a hammer on pegs, but having a charged battery could enable a secondary path to that location). You could leave one battery permanently powering a central teleport and remove batteries from distant teleports you don't need to access anymore, or a second player could take the battery from the central teleport and use it on a different distant one, so the first player would travel to the second player's location once she uses the first distant teleport.
- batteries could be used to power something like an automated medical bed that heals a player
- batteries could also be used as the ultimate progression item, i.e. the game would require you to power up a number of distinct gizmos at the same time to make the final boss area accessible.

I think this battery gameplay could be the distinguishing factor for TetraForce.  ALTTP has the light and dark world, OOC and Skyward Sword have the two timelines, Spirit Tracks the trains... TetraFroce could mean four power packs that need to be acquired, but they would be more useful than simply collecting crystals, instruments or other rather single use items.

**Comment by Nodding** [comment]
After reading it all I agree an electric dungeon would be really neat. I think to avoid too much new stuff at once I would stick to having this be intended for later game (though accessible early of course) and stick to the idea of having four batteries spread through out the dungeon that need to be placed to power a door for the dungeon's boss. There would be generators, wires, mechanical doors /bridges/ teleport pads like you said that need power etc. 

I would want to make sure we also balance the dungeon for single players and multiplayer in terms of puzzles needing more than one set of hands. I wouldn't mind a section being easier with more people, but it shouldn't be impossible without them. If a room or two were we should at least have a sign or something warning the players you need more hands.

As for the dungeon's item, I could see insulated gloves or something making you able to handle high capacity batteries, but I could also see a portable battery pack you could charge and empty into a generator or something... this could then be easily implemented into the over world for secrets and goodies hidden with more power puzzles.

I'm unsure about using the batteries as distinct items of the game, as I believe planned now the idea of the game is revolving around the players collecting items stolen from them from the thieves guild, in addition to collecting ingredients to save the queen of the land for the king. This is all loose and barely talked about or touched upon so further discussion with others is definitely needed.

**Reply by IQGobo** [reply]
I initially thought the batteries would act as boss loot from four different dungeons, but collecting 4 of them in a single dungeon could work as well. This would require the dungeon to be quite big as well and maybe it would turn the dungeon into some kind of hub area. Like, you have access to different wings of the dungeon (each section containing one battery), but each wing requires another progression item to complete. You could visit the electric dungeon several times, after getting hold of each item individually, or you could clear it in one go after collecting all the progression items needed for completion.

My reasoning was, that acquiring a first battery from one dungeon would give the player the benefit of gambling for options: will I use my charge on this bridge to access another area? Will I put it into this teleport with easy access to other areas of the game to help me run some other errands quicker? Will I use it to open a door to a place that has three chests that might contain useful loot?
Then later on, you either get other batteries as well or at least gain access to a charging station. And by that you would have access to all the optional or alternative paths and locations.

My main goal with this system was to add choice and alternative routing to the world map. It would feel better getting something useful from killing a boss, compared to just getting pretty pendants or crystals, just to make use of them once throughout the game.

I forgot about the thief guild background, thanks for reminding me!
